### ISSUE TITLE
Advance training technics used

### ISSUE THREAD

**Issue by algila** [nejlag/Semi-Supervised-Learning-GAN#3]
Thanks for sharing.
May you kindly list the advance training technics you implemented from article "Improved Techniques for Training GANs" ?
Did you implemented all the below ? Have you compared what of those was more effective improving the final score ?
1) Feature matching
2) Minibatch discrimination (the article suggest to do not use for the need of semi-supervised-learning 
3) Historical averaging
4) One-sided label smoothing
5) Virtual batch normalization

Reading the article it is not clear to me if they used all or not. Do you have a better understanding ?

**Comment by nejlag** [comment]
Hi,

I implemented **feature matching** and **one-sided label smoothing** and I didn't see any improvement using **one-sided label smoothing**. So, my code only has **feature matching**. But, I think they used all proposed techniques in their evaluations (look at Fig. 6 description for example: _(Right) Samples generated using the techniques proposed in this work_).

The other thing that I tried which slightly helped was the **pull-away term** proposed by (https://arxiv.org/pdf/1705.09783.pdf ) not with MNIST data though....

**Comment by algila** [comment]
Hi, 
you combined the feature matching cost function with the known generator cost that maximizes the log-probability of discriminator. Apper to me logical, but reading the article doesn't Apper evident to be that they implemented as you did. 
Have you tested the case using generator cost function having only the feature matching term ?

**Comment by nejlag** [comment]
Yes, the paper uses feature_matching only and not the known generator cost. I combined the two myself as I mentioned in the README file. As paper says, with feature_matching you don't get good quality generated samples. When I combined the two losses, I got good quality generated samples as well as good results for the discriminator! I should say though that the other paper that I mentioned in my previous comment says that a good semi-supervised learning requires a bad GAN! So, my experience is in contrast with that paper! But, I didn't investigate it further....
